app/forms/webauthn_setup_form.rb

```diff
@@ -46,13 +47,13 @@ def platform_authenticator?
 
   attr_reader :success, :transports, :invalid_transports
   attr_accessor :user, :challenge, :attestation_object, :client_data_json,
-                :name, :platform_authenticator, :authenticator_data_flags
+                :name, :platform_authenticator, :authenticator_data_flags, :nice_name
 
   def consume_parameters(params)
     @attestation_object = params[:attestation_object]
     @client_data_json = params[:client_data_json]
-    @name = params[:name]
     @platform_authenticator = (params[:platform_authenticator].to_s == 'true')
+    @name = !@platform_authenticator ? params[:name] : @device_name
```

Comment:
might be clearer to flip this so we check a positive condition
```suggestion
    @name = @platform_authenticator ? @device_name : params[:name]
```